src/components/manifold-subscription-create/manifold-subscription-create.tsx

```diff
@@ -205,36 +215,44 @@ export class ManifoldSubscriptionCreate {
         },
       });
 
-      this.subscribing = false;
-
       if (error) {
+        this.subscribing = false;
         this.setupIntentError = error.message;
-      } else {
-        this.setupIntentStatus = setupIntent?.status;
-        if (setupIntent?.status === 'succeeded') {
-          const configuredFeatures = Object.keys(this.configuredFeatures).map(key => ({
-            label: key,
-            value: `${this.configuredFeatures[key]}`,
-          }));
-
-          const variables: CreateSubscriptionMutationVariables = {
-            ownerId: undefined,
-            planId: this.planId,
-            configuredFeatures,
-          };
-
-          const res = await this.connection?.graphqlFetch<CreateSubscriptionMutation>({
-            query: createSubscrptionMutation,
-            variables,
-          });
-
-          // eslint-disable-next-line no-console
-          console.log({ res });
+        return;
+      }
+
+      this.setupIntentStatus = setupIntent?.status;
+      if (setupIntent?.status === 'succeeded' && this.connection) {
+        const configuredFeatures = Object.keys(this.configuredFeatures).map(key => ({
+          label: key,
+          value: `${this.configuredFeatures[key]}`,
+        }));
+
+        const variables: CreateSubscriptionMutationVariables = {
+          ownerId: undefined,
+          planId: this.planId,
+          configuredFeatures,
+        };
+
+        const { data, errors } = await this.connection.graphqlFetch<CreateSubscriptionMutation>({
+          query: createSubscrptionMutation,
+          variables,
+        });
+
+        if (errors) {
+          // TODO handle errors
+        }
+
+        if (data) {
+          this.createSuccess.emit({ id: data.createSubscription.data.id });
         }
       }
+      this.subscribing = false;
```

Comment:
Just a note: when I tried to subscribe the 2nd time, nothing happened—no error state or anything. Should there be some “finish” state when successful (if yes, probably not a part of this PR, but still wanted to ask)?

Reply:
We decided to just let the developer handle the success state by listening for the event. I think it should probably reset though and allow another subscription once it's done with the first one